README.md

```diff
@@ -66,55 +66,65 @@ flowchart LR
+is to embed the C shared library using `go:embed`. Then to dump it into a file, load it using `dlopen` and to load the
+symbols using `dlsym`. And finally to call them.
 
 ⚠️ Keep in mind that **purego only works on linux/darwin for amd64/arm64 and so does go-libddwaf.**
 
-Another requirement of `libddwaf` is to have a FHS filesystem on your machine and, for linux, to provide `libc.so.6`, `libpthread.so.0` and `libm.so.6`, `libdl.so.2` as dynamic libraries.
+Another requirement of `libddwaf` is to have a FHS filesystem on your machine and, for linux, to provide `libc.so.6`,
+`libpthread.so.0` and `libm.so.6`, `libdl.so.2` as dynamic libraries.
 
-## Contributing usual pitfalls
+## Contributing pitfalls
 
-- Cannot dlopen twice in the app lifetime on OSX
-- `runtime.KeepAlive()` calls are here to prevent the GC from destroying objects too early
-- Since there is a stack switch between the go code and the C code, usually the only C stacktrace you will ever get is from gdb
-- If a segfault happens during a call to the C code, the goroutine stacktrace which has done the call is the one annotated with `[syscall]`.
+- Cannot dlopen twice in the app lifetime on OSX. It messes with Thread Local Storage and usually finishes with a `std::bad_alloc()`
+- `keepAlive()` calls are here to prevent the GC from destroying objects too early
+- Since there is a stack switch between the go code and the C code, usually the only C stacktrace you will ever get is from GDB
+- If a segfault happens during a call to the C code, the goroutine stacktrace which has done the call is the one annotated with `[syscall]`
 - [GoLand](https://www.jetbrains.com/go/) does not support `CGO_ENABLED=0` (as of June 2023)
-- Keep in mind that we fully escape the type system. If you send the wrong data it will segfaults in the best cases but not always!
-- The structs in `ctypes.go` are here to reproduce the memory layout of the structs in `include/ddwaf.h` because pointer to these structs will be passed directly.
+- Keep in mind that we fully escape the type system. If you send the wrong data it will segfaults in the best cases but not always !
+- The structs in `ctypes.go` are here to reproduce the memory layout of the structs in `include/ddwaf.h` because pointer to these structs will be passed directly
 - Do not use `uintptr` as function arguments or results types, coming from `unsafe.Pointer` casts of Go values, because they escape the pointer analysis which can create wrongly optimized code and crash. Pointer arithmetic is of course necessary in such a library but must be kept in the same function scope.
+- GDB is available on arm64 but is not officially supported so it usually crashes pretty fast
+- No pointer from variables on the stack shall be sent to the C code because Go stacks can be moved during the C call. More on this [here](https://medium.com/@trinad536/escape-analysis-in-golang-fc81b78f3550)
```

Comment:
```suggestion
- No pointer to variables on the stack shall be sent to the C code because Go stacks can be moved during the C call. More on this [here](https://medium.com/@trinad536/escape-analysis-in-golang-fc81b78f3550)
```